libsupport/include/katana/CommBackend.h

```diff
@@ -36,7 +36,9 @@ class KATANA_EXPORT CommBackend {
   /// The number of tasks involved
   uint32_t Num{1};
   /// The id number of this task
-  uint32_t ID{0};
+  uint32_t Rank{0};
+  /// The local rank of this task (process ordinal number within within its machine)
+  uint32_t LocalRank{0};
```

Comment:
I know I'm late to the party. But:

Why "rank" instead of "host"? Everywhere else in the Katana we are calling them hosts. This seems inconsistent. Is there a specific reason for using the MPI term here, but not elsewhere?

Reply:
Sorry, @arthurp , I didn't plan to merge the PR that quickly initially, but then I started needing it for benchmarking. Host would be wrong as a concept, because a host is a machine usually, which may contain multiple processes. The rank is the ordinal number of the process, not the machine/host. And the local rank would not be compatible with the term "host" at all.

Reply:
I kinda agree. But then we need to change to something like "rank" everywhere not just here. Right now, "host" is used instead of "rank" everywhere else in the code base. Consistency is very important. I'm OK with a search and replace to move to using something other than "host", but I'm not OK with a mix of terms.

Reply:
We should avoid using `host` because it's the process ordinal number, not a host ordinal number within the system. I think search and replace wouldn't help here, because the word host is overloaded and used in many contexts in the codebase. Sometimes it really means a machine, sometimes it means the MPI process. In https://github.com/KatanaGraph/katana-enterprise/pull/2039 I changed the places where `CommBackend::ID` was used. But for `host` we would have to look case-by-case.

Reply:
In which cases does `host` mean machine (only) and not MPI process?

It seems strange to me that any code would make such a distinction as we use MPI for communication and MPI abstracts hosts from processes.

Reply:
Disagree with Serge. We've always used host to mean a process, not a machine. I'm not sure if that's standard. 

For heterogeneous execution, we use hosts or devices interchangeably. Each device has its own host process, so its still consistent with our use of host as a process. 

I liked `ID` over `Rank` to indicate the network/communication identifier. However, once Serge introduced "local" network identifier within that machine, `ID` can be confusing, especially because we have so many IDs already. 

I think we're conflating host and ID/rank. Host is a process (or unit of communication). Host ID/rank is its (communication) identifier. We explicitly say HostID in the code in many places. I don't think there is a break in consistency b/w host and rank. The break in consistency is b/w ID and Rank; we have HostID, not HostRank. However, that's arguable. I'd say that the type is still `HostID` for both `Rank` and `LocalRank` variables, which capture different network/communication identifiers.